Automattic/woocommerce-payments
# Add charge ID to order meta for transaction listing

Fixes #1803 

#### Changes proposed in this Pull Request
Previously I was not receiving web hook events for Giropay transactions. Upon speaking with Stripe I was then seeing these web hook events firing and thus was able to get Giropay transactions displayed in the Payments > Transactions page. While viewing this page I had noticed that the Order ID was not populated for Giropay transactions. This PR address that issue by adding the charge ID to the order meta data to create that relationship. 

Given the short lived nature of this gateway, this particular change is more to confirm we are capturing payment details after a redirect correctly.



#### Testing instructions
1. Start with a store using Euros and [Giropay enabled](https://github.com/Automattic/woocommerce-payments/pull/1577).
2. Place an order and pay with Giropay.
3. Upon completion the transaction should be appearing in the Payments > Transactions page.
4. The transaction listing should include a linked Order ID.
5. The transaction detail should also include the linked Order ID.

**Before**
![Giropay-Before](https://user-images.githubusercontent.com/68524302/118875518-23dc4c80-b8ba-11eb-9541-708bc2c3f9cf.png)

**After**
![Giropay-After](https://user-images.githubusercontent.com/68524302/118875542-29399700-b8ba-11eb-884c-d48d40e724aa.png)

-------------------


**Notes**
You'll notice the Source field is also empty on these transactions. This is tracked in an issue on the Payments Server repo.
